@navikt/core/react/src/overlays/floating-menu/Menu.tsx

```diff
@@ -0,0 +1,1192 @@
+    const composedRefs = useMergeRefs(forwardedRef, ref);
+    const isPointerDownRef = useRef(false);
+
+    const handleSelect = () => {
+      const menuItem = ref.current;
+      if (!disabled && menuItem && onSelect) {
+        const itemSelectEvent = new CustomEvent(ITEM_SELECT_EVENT, {
+          bubbles: true,
+          cancelable: true,
+        });
+        menuItem.addEventListener(ITEM_SELECT_EVENT, onSelect, { once: true });
+        /* dispatchDiscreteCustomEvent */
+        ReactDOM.flushSync(() => menuItem.dispatchEvent(itemSelectEvent));
+        if (itemSelectEvent.defaultPrevented) {
+          isPointerDownRef.current = false;
+        } else {
+          rootContext.onClose();
+        }
+      }
+    };
+
+    return (
+      <MenuItemImpl
+        {...rest}
+        tabIndex={disabled ? -1 : 0}
+        ref={composedRefs}
+        disabled={disabled}
+        onClick={composeEventHandlers(onClick, handleSelect)}
+        onPointerDown={(event) => {
+          onPointerDown?.(event);
+          isPointerDownRef.current = true;
+        }}
+        onPointerUp={composeEventHandlers(onPointerUp, (event) => {
+          // Pointer down can move to a different menu item which should activate it on pointer up.
+          // We dispatch a click for selection to allow composition with click based triggers and to
+          // prevent Firefox from getting stuck in text selection mode when the menu closes.
+          if (!isPointerDownRef.current) event.currentTarget?.click();
+        })}
+        onKeyDown={composeEventHandlers(onKeyDown, (event) => {
+          if (disabled) {
+            return;
+          }
+          if (SELECTION_KEYS.includes(event.key)) {
+            event.currentTarget.click();
+            /**
+             * We prevent default browser behaviour for selection keys as they should only trigger
+             * selection.
+             * - Prevents space from scrolling the page.
+             * - If keydown causes focus to move, prevents keydown from firing on the new target.
+             */
+            event.preventDefault();
+          }
+        })}
+      />
+    );
+  },
+);
+
+type MenuItemImplElement = SlottedDivElementRef;
+
+interface MenuItemImplProps extends SlottedDivProps {
+  disabled?: boolean;
+}
+
+const MenuItemImpl = forwardRef<MenuItemImplElement, MenuItemImplProps>(
+  (
+    {
+      disabled = false,
+      onPointerMove,
+      onPointerLeave,
+      onFocus,
+      onBlur,
+      ...rest
+    }: MenuItemImplProps,
+    forwardedRef,
+  ) => {
+    const { register } = useMenuDescendant({ disabled });
+
+    const contentContext = useMenuContentContext();
+    const ref = useRef<HTMLDivElement>(null);
+    const composedRefs = useMergeRefs(forwardedRef, ref, register);
+    const [isFocused, setIsFocused] = useState(false);
+
+    return (
+      <SlottedDivElement
+        role="menuitem"
+        data-highlighted={isFocused ? "" : undefined}
```

Comment:
Do we need this? `data-highlighted`

Reply:
Hmm not really, think a simple `:focus`-check with CSS will be enough. removed it + `isFocused`